frontend/src/components/ui/StationListWindow/hooks/useInfiniteStationSummaries.ts

```diff
@@ -0,0 +1,64 @@
+import { useEffect } from 'react';
+
+import { useInfiniteQuery, useQueryClient } from '@tanstack/react-query';
+
+import { fetchStationSummaries } from '@hooks/fetch/fetchStationSummaries';
+
+import { cachedStationSummariesActions } from '@ui/StationListWindow/tools/cachedStationSummaries';
+
+import { QUERY_KEY_STATION_SUMMARIES } from '@constants/queryKeys';
+
+import type { StationMarker, StationSummary } from '@type';
+
+interface StationSummaryResponse {
+  stations: StationSummary[];
+  nextPage: number;
+}
+
+const makeStationIdsChunks = (markers: StationMarker[]) => {
+  return markers.reduce((acc: string[][], marker, index) => {
+    const REQUEST_CHUNK_SIZE = 10;
+    const chunkIndex = Math.floor(index / REQUEST_CHUNK_SIZE);
+
+    if (!acc[chunkIndex]) {
+      acc[chunkIndex] = [];
+    }
+
+    acc[chunkIndex].push(marker.stationId);
+
+    return acc;
+  }, []);
+};
+
+export const useInfiniteStationSummaries = (markers: StationMarker[]) => {
+  const queryClient = useQueryClient();
+  const stationIdChunks = makeStationIdsChunks(markers);
+
+  useEffect(() => {
+    queryClient.removeQueries([QUERY_KEY_STATION_SUMMARIES]);
+  }, [markers]);
+
+  return useInfiniteQuery<StationSummaryResponse>(
+    [QUERY_KEY_STATION_SUMMARIES],
+    async ({ pageParam = 0 }) => {
+      const stationIds = stationIdChunks[pageParam] ?? [];
+      const uncachedStationSummaryIds = stationIds.filter(
+        (stationId) => !cachedStationSummariesActions.has(stationId)
+      );
+
+      if (uncachedStationSummaryIds.length > 0) {
+        const stationSummaries = await fetchStationSummaries(uncachedStationSummaryIds);
+        cachedStationSummariesActions.add(stationSummaries);
+        return { stations: stationSummaries, nextPage: pageParam + 1 };
+      } else {
+        return { stations: [], nextPage: -1 };
+      }
```

Comment:
앞선 리뷰에서 말씀 드린 것과 같이 이 부분에서도 요청을 넣기 직전에 캐시 데이터에 stationId가 존재하는지 여부를 판별하지 않고 stationIdChunks를 생성할 때 이를 미리 제거한 상태로 그냥 stationIds를 통해 요청을 발생시키는 것이 좋을 것 같습니다!

이렇게 되면 이전에 포기했던 10개씩 새로운 충전소 목록을 불러오는 것이 가능해질 것 같습니다!

```suggestion
      const stationIds = stationIdChunks[pageParam] ?? [];

      if (stationIds.length > 0) {
        const stationSummaries = await fetchStationSummaries(stationIds);
        cachedStationSummariesActions.add(stationSummaries);
        return { stations: stationSummaries, nextPage: pageParam + 1 };
      } else {
        return { stations: [], nextPage: -1 };
      }
```

Reply:
이해했습니다. 반영완료했어요